src/ChatManager.js

```diff
@@ -1,7 +1,37 @@
 
+function parseJwt(token) {
+  var base64Url = token.split('.')[1];
+  var base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
+  var jsonPayload = decodeURIComponent(atob(base64).split('').map(function(c) {
+      return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
+  }).join(''));
+
+  return JSON.parse(jsonPayload);
+};
+
+function parseUserName(token) {
+  const sub = parseJwt(token)['sub']
+  const startChar = sub.indexOf('|');
+  const endChar = sub.indexOf('@');
```

Comment:
per our conv, if desired, perhaps separately from this change, dablchat could in principle exchange the access token received after a successful authentication to load details about the authenticated user via https://github.com/DACH-NY/daml-cloud/blob/f032b533cd774d86ff4c4dedd984432bf7a2f3b4/dabl-services/dabl_svc/site_api/user.py#L13

Reply:
yes. I am not fond of that `parseUserName` function. Let's tackle it once we are done with the dabl side of things